projects/adc_challenge/src/main.c

```diff
@@ -0,0 +1,63 @@
+#include "adc.h"
+#include "gpio.h"
+#include "gpio_it.h"
+#include "interrupt.h"
+#include "log.h"
+#include "soft_timer.h"
+#include "wait.h"
+
+// reads from the specified address and logs the value
+static void prv_read_from_adc_addr(const GpioAddress *address, void *context) {
+  GpioAddress addr = {
+    .port = address->port,  //
+    .pin = address->pin     //
+  };
+  uint16_t adc_data = 0;
+  adc_read_converted_pin(addr, &adc_data);
+
+  LOG_DEBUG("ADC pin has value %d", adc_data);
+}
+
+int main() {
+  interrupt_init();
+  soft_timer_init();
+  gpio_init();
+  gpio_it_init();
+
+  GpioAddress input_addr = { .port = GPIO_PORT_A, .pin = 6 };
```

Comment:
Please add a comma beside 6
`GpioAddress input_addr = { .port = GPIO_PORT_A, .pin = 6, };`
Then once you run make format, you will get a similar format to your input_settings struct